src/test/java/org/opensearch/security/auditlog/compliance/RestApiComplianceAuditlogTest.java

```diff
@@ -41,21 +44,23 @@ public void testRestApiRolesEnabled() throws Exception {
             .build();
 
         setup(additionalSettings);
-        TestAuditlogImpl.clear();
-        String body = "{ \"password\":\"some new password\",\"backend_roles\":[\"role1\",\"role2\"] }";
-        HttpResponse response = rh.executePutRequest(
-            "_opendistro/_security/api/internalusers/compuser?pretty",
-            body,
-            encodeBasicHeader("admin", "admin")
-        );
-        Thread.sleep(1500);
-        Assert.assertEquals(HttpStatus.SC_CREATED, response.getStatusCode());
-        Assert.assertTrue(TestAuditlogImpl.messages.size() + "", TestAuditlogImpl.messages.size() == 1);
-        Assert.assertTrue(TestAuditlogImpl.sb.toString().contains("audit_request_effective_user"));
-        Assert.assertFalse(TestAuditlogImpl.sb.toString().contains("COMPLIANCE_INTERNAL_CONFIG_READ"));
-        Assert.assertTrue(TestAuditlogImpl.sb.toString().contains("COMPLIANCE_INTERNAL_CONFIG_WRITE"));
-        Assert.assertTrue(TestAuditlogImpl.sb.toString().contains("UPDATE"));
-        validateMsgs(TestAuditlogImpl.messages);
+        Thread.sleep(1000);
```

Comment:
Is this sleep needed?

Reply:
I added `sleep` after every instance of `setup` to ignore the first few audit messages generated on setup. The only exception is for the testAutoInit test where it is specifically testing for audit messages generated on startup.

Reply:
> I added sleep after every instance of setup to ignore the first few audit messages generated on setup.

This is still a timing bug waiting to happen - maybe not on your machine but in the GH Runners.  If there are still messages that can pollute the tests - and are expected could setup use `doThenWaitForMessages(...)` to ensure everything is stable?

Reply:
Good idea. I updated this to not rely on sleep and instead wait for the expected messages to be received.